### PR TITLE
Extend the benchmarking possibilities of libmata

### PR DESCRIPTION
**Pull request by tfiedor**
This PR:
  1. Adds nfa-bench as a submodule for more smooth benchmarking
  2. Adds binaries for reproducing CADE'23 results (to certain degree)
  3. Adds inputs and jobs for nfa-bench stuff.

### CODE REVIEW

**Comment by tfiedor** [comment]
I tagged most of you as reviewers, but I'd say it should suffice if @Adda0 and maybe @vhavlena checks this out. There might be some issues for macboys.

File: tests-integration/README.md [hunk 1]
```diff
@@ -61,3 +61,25 @@ In order to test or measure performance of the code you just built by the above
     `jobs/corr-single-param-jobs.yaml` (for binaries accepting single argument) and 
     `jobs/corr-double-param-jobs.yaml` (for binaries accepting two arguments).
   5. This will generate `csv` report of measuring binaries registered in `jobs/*.yaml` files on automata listed in the `input/*.input` files.
+ 
+### How to benchmark and reproduce some of our results
+
+We assume, this is executed from `tests-integration` directory.
```
Comment (Adda0) [inline tests-integration/README.md:67]: Typo.
```suggestion
We assume this is executed from `tests-integration` directory.
```

File: tests-integration/scripts/process_pyco.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    echo "$1"
+  else
+    echo "$2/$1"
+  fi
+}
+
+processed_header=false
+
+
+# For each benchmakr
```
Comment (Adda0) [inline tests-integration/scripts/process_pyco.sh:71]: Typo.
```suggestion
# For each benchmark
```

**Review by Adda0 — APPROVED** [review]
Fix the following warning, please:
```
.../mata/tests-integration/src/bench-bool-comb-intersect.cc: In function ‘int main(int, char**)’:
.../mata/tests-integration/src/bench-bool-comb-intersect.cc:39:48: warning: conversion to ‘std::vector<Mata::Nfa::Nfa>::size_type’ {aka ‘long unsigned int’} from ‘int’ may change the sign of the result [-Wsign-conversion]
   39 |         result = intersection(result, automata[i]);
      |                                                ^
```

Add regexes to `.gitignore` to ignore files generated by intergration tests such as the following:
```
?? tests-integration/inputs/bench-double-automata-inclusion.input
?? tests-integration/inputs/bench-double-bool-comb-cox.input
?? tests-integration/inputs/bench-quintuple-email-filter.input
?? tests-integration/inputs/bench-single-z3-noodler.input
?? tests-integration/inputs/bench-variadic-bool-comb-ere.input
?? tests-integration/inputs/bench-variadic-bool-comb-intersect.input
?? tests-integration/inputs/double-automata.input
?? tests-integration/inputs/single-automata.input
?? tests-integration/jobs/bench-cade-23.yaml
?? tests-integration/jobs/corr-double-param-jobs.yaml
?? tests-integration/jobs/corr-single-param-jobs.yaml
?? tests-integration/jobs/corr-test-jobs.yaml
?? tests-integration/results/
?? tests-integration/src/concatenate_remove-epsilon-symbols.cc
?? tests-integration/src/move-semantics.cc
?? tests-integration/src/utils/config.hh
```

Earlier, we talked about having a folder `/automata/` in the project root with all benchmarks, automata instances, etc. For now, we have `/tests-integration/automata` and `tests-integration/nfa-bench`. What do you think about encapsulating all benchmarks, automata instances, submodules with benchmarks and all else in a single root folder? 

Otherwise, the PR looks good to me. The integration tests from the examples in `README.md` run, even if they are killing my notebook.

**Comment by Adda0** [comment]
Reviews will be resolved in another PR. Pending reviewers are welcome to look at this PR.

**Pull request merged**